src/components/DataStructures/Graph/GraphTracer.js

```diff
@@ -379,26 +379,32 @@ class GraphTracer extends Tracer {
     recursiveAnalyze(root, 0);
 
     // Calculates node's x and y.
-    const hGap = rect.width / leafCounts[root];
-    const vGap = rect.height / maxDepth;
     marked = {};
-    const recursivePosition = (node, h, v) => {
+    const recursivePosition = (node, h, v, x, y) => {
       marked[node.id] = true;
-      node.x = rect.left + (h + leafCounts[node.id] / 2) * hGap;
-      if (this.nodes.length === 2 && h === 0 && v === 1) {
-        node.x = rect.left + (h + leafCounts[node.id] / 4) * hGap;
-      }
-      node.y = rect.top + v * vGap;
+      node.x = x;
+      node.y = y;
       const linkedNodes = this.findLinkedNodes(node.id, false);
       if (sorted) linkedNodes.sort((a, b) => a.id - b.id);
       for (const linkedNode of linkedNodes) {
         if (marked[linkedNode.id]) continue;
-        recursivePosition(linkedNode, h, v + 1);
+        let x1 = x;
+        let y1 = y;
+        // For left child
+        if (linkedNode.id === 2 * node.id) {
+          x1 -= leafCounts[node.id] * 30;
```

Comment:
could this be commented? I'm not sure what the numbers mean here

Reply:
Actually we can't ..  For each leafnodes we are assigning 30. If we remove that the tree will become vertical.

Reply:
Ah - not commented out, commented as in explained - or the numbers could be factored out into const declarations with descriptive names?

Reply:
Added constants and comments regarding two variables